rewrite-yaml/build.gradle.kts

```diff
@@ -26,4 +26,5 @@ dependencies {
     implementation("io.micrometer:micrometer-core:1.9.+")
 
     testImplementation(project(":rewrite-test"))
+    testImplementation("org.junit-pioneer:junit-pioneer:2.0.0")
```

Comment:
So far we've mostly used pinned: https://github.com/search?q=org%3Aopenrewrite%20junit-pioneer&type=code
So I thought to keep it consistent in this case, since we span multiple repositories of this usage as well.
Open to standardize on something else, but that's then probably best as a separate effort everywhere.